sound/soc/sof/ops.h

```diff
@@ -387,10 +387,47 @@ static inline int snd_sof_dma_trace_release(struct snd_sof_dev *sdev)
 
 static inline int snd_sof_dma_trace_trigger(struct snd_sof_dev *sdev, int cmd)
 {
-	if (sof_ops(sdev)->trace_trigger)
-		return sof_ops(sdev)->trace_trigger(sdev, cmd);
+	int ret = 0;
 
-	return 0;
+	if (!sdev->dtrace_is_supported)
+		return 0;
+
+	/* FIXME: return 0 or return -EINVAL ? */
+	if (!sof_ops(sdev)->trace_trigger)
+		return 0;
+
+	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_START:
+		/* FIXME: return 0 when dtrace_is_enabled? */
```

Comment:
what is this fixme intended for? whether you should return 0 or error?

Reply:
I mean whether we should return other value instead of "0" to tell the caller dma trace is already enabled. I will remove this comment.